### PR TITLE
[Concept Entry] Ai Neural Networks (ANN)

### PR DESCRIPTION
**Pull request by DevGat718**
### Description

-updated a few subject entries 

### Type of Change

- Updating the documentation

### Checklist

- [ x] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by SSwiniarski** [comment]
@DevGat718, you should update subjects.md and tags.md in the same PR as the entry that is using the new entries. Fortunately, because you branched off of your entry for the other PRs (which is not what you're supposed to do, for a new PR you should always branch off of main, but in this case it worked out.) this PR has all the files updated. This is the only one that needs to be open, so I'm closed the others as duplicates. I also updated the title of this PR to be more descriptive.

**Comment by SSwiniarski** [comment]
@DevGat718 you also need to remember to sign the [CLA](https://cla-assistant.io/Codecademy/docs?pullRequest=2190).

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:5]: The subjects you've added are more appropriate as tags. This is the suggested subjects for this entry:
```
  - 'Machine Learning'
  - 'Computer Science'
```

File: documentation/tags.md
Comment (SSwiniarski) [inline documentation/tags.md:16]:
```suggestion
ANN
```

File: documentation/tags.md
Comment (SSwiniarski) [inline documentation/tags.md:199]:
```suggestion
```
This is duplicative of Neural Networks, and is less precise.

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:16]:
```suggestion
```
This tag is duplicative of Neural Networks, and is less precise.

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:22]:
```suggestion
**Artificial neural networks (ANN)** are a method in artificial intelligence that teaches computers to process data in a way that is inspired by the human brain. 

```

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:26]:
```suggestion
- An ANN is one of the main tools used for machine learning, it is a brain-inspired system used to replicate the way that humans learn.
```
We don't like first or second person pronouns in Docs.

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:30]:
```suggestion
- Neural networks are trained on specific data which then get broken down into the input layer (where information is fed into) then to the hidden layer (where weight & bias are calculated from the input layer) and lastly the output layer (where information is output after algorithms are processed from the hidden and input layers) via perceptrons (similar to human neurons but for machines to process, store, calculate & move data forward)
```

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:34]:
```suggestion
### AI
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@DevGat718 I did a first review and left you comments. As you make changes, make sure you are making them to the "subjects" branch, since that is the branch that has the updates on all three files, and the one connected to this PR.

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:5]: @SSwiniarski copy that, should I deleted the rest of the subjects & leave only those 2? 
And also is it ok if I delete the other 2 branches?

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:16]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:22]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:26]: copy, done.

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:30]: done but what about the & between (weight & bias)?

Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:30]: also (calculate & move data)?

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:34]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:39]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (SSwiniarski) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:30]: @DevGat718 Yes, I missed those.

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (HishamT) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:41]:
```suggestion
Neural networks are the core of machine learning algorithms. A neural network is trained on specific datasets from which it extracts the information and passes it through different layers searching for patterns.
```

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (HishamT) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:25]:
```suggestion
An ANN is a type of machine learning process, called deep learning, that uses interconnected nodes or neurons in a layered structure that resembles the human brain.
```

**Review by HishamT — CHANGES_REQUESTED** [review]
Finished reviewing and left a few comments. I will review again once all of @SSwiniarski's comments are addressed. One thing I did notice, however, the first definition of an ANN is exactly copied from [here](https://aws.amazon.com/what-is/neural-network/).

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:22]: @HishamT ill re-word it.

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:30]: copy im on it. @SSwiniarski

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:41]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:42]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:36]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:37]: done

File: documentation/subjects.md
Comment (DevGat718) [reply documentation/subjects.md:1]: which ones exactly? @SSwiniarski

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:41]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:25]: done

File: documentation/tags.md
Comment (DevGat718) [reply documentation/tags.md:205]: done

File: documentation/tags.md
Comment (DevGat718) [reply documentation/tags.md:16]: done

File: documentation/subjects.md
Comment (DevGat718) [reply documentation/subjects.md:1]: nvm I figured it out.

**Comment by DevGat718** [comment]
trying to push my changes back in but im getting these errors, help.

Screenshot 2023-05-19 at 12 47 41 AM

**Comment by DevGat718** [comment]
oh wait do I need to fetch? will that mess with my changes I made?

**Comment by SSwiniarski** [comment]
You should be able to do a git pull.

**Comment by DevGat718** [comment]
its not letting me push my updated code to here, 

Screenshot 2023-05-21 at 5 52 05 AM

**Comment by DevGat718** [comment]
Screenshot 2023-05-21 at 6 05 18 AM

this what im getting on the git pull

**Comment by SSwiniarski** [comment]
@yangc95, @KTom101, @caupolicandiaz, @HishamT: Any of you have any suggestions to help @DevGat718?

**Comment by HishamT** [comment]
Hey @DevGat718, have you tried rebasing by using `git rebase`?

**Comment by yangc95** [comment]
@DevGat718 Can you try pulling without fast-forward? `git pull --no-ff`

**Comment by DevGat718** [comment]
I'll try the pulling w/o fast-forwarding 1st @yangc95 if that don't work ill try using git rebase good idea! @HishamT

**Comment by DevGat718** [comment]
I resolved it. lmk if any more updates are needed

**Comment by DevGat718** [comment]
I realized what I was doing wrong, I was making changes on top of your changes not realizing you made changes my bad! lol smh

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (SSwiniarski) [inline content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:13]:
```suggestion
```
I don't think Brain Activity is a great tag, as it's only tangential to computer science.

File: documentation/tags.md
Comment (SSwiniarski) [inline documentation/tags.md:16]:
```suggestion
```
An ANN tag is duplicative with "Neural Networks."

File: documentation/tags.md
Comment (SSwiniarski) [inline documentation/tags.md:45]:
```suggestion
Break
```
I don't think Brain Activity is a great tag, as it's only tangential to computer science.
(And you shouldn't be deleting existing tags.)

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@DevGat718 I reviewed your changes and left a few more comments. Just some typos and a couple of tags I didn't agree with.

File: documentation/tags.md
Comment (DevGat718) [reply documentation/tags.md:45]: that was a accident I had merge conflict w/ brain activity when I resolved it it got deleted.

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:29]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:41]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:10]: done

File: content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/terms/artificial-neural-networks/artificial-neural-networks.md:13]: done

File: documentation/tags.md
Comment (DevGat718) [reply documentation/tags.md:16]: done

File: documentation/tags.md
Comment (DevGat718) [reply documentation/tags.md:45]: done

**Comment by DevGat718** [comment]
Done

**Comment by yangc95** [comment]
@DevGat718 I believe this is the concept entry for Neural Networks. Could you restructure this PR by taking the file out of the terms folder and placing it right into the `neural-networks` folder? So the complete path would be: `docs/content/ai/concepts/neural-networks/neural-networks.md`

cc: @HishamT

**Comment by DevGat718** [comment]
@yangc95 that path was given to me in #2015 but im having issues removing it from that folder if that's what you want me to do.

**Comment by SSwiniarski** [comment]
@DevGat718 I moved the entry for you, per: @yangc95, and also did two other small edits.

**Review by SSwiniarski — APPROVED** [review]
@DevGat718 Looks good now. Let's see what @HishamT thinks.

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (HishamT) [inline content/ai/concepts/neural-networks/neural-networks.md:34]:
```suggestion
AI works on a collection of technologies that allow the computer to sense, learn, reason & act.
```

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (HishamT) [inline content/ai/concepts/neural-networks/neural-networks.md:39]:
```suggestion
Neural networks are the core of deep learning algorithms. A neural network is trained on specific datasets from which it extracts the information and passes it through different layers searching for patterns.
```

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (HishamT) [inline content/ai/concepts/neural-networks/neural-networks.md:40]:
```suggestion
Once a neural network is trained on a dataset, it can start to encounter new scenarios and start making predictions based on that previous dataset. 
```

**Review by HishamT — CHANGES_REQUESTED** [review]
Just some small changes to be made, and then it should be good.

**Comment by SSwiniarski** [comment]
@DevGat718 I went through and did an `npm run format` so the linting checks pass and we can merge right after @HishamT's comments are addressed and approved.

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/neural-networks.md]: done

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/neural-networks.md:24]: done

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/neural-networks.md:28]: done

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/neural-networks.md:34]: done

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/neural-networks.md:39]: done

File: content/ai/concepts/neural-networks/neural-networks.md
Comment (DevGat718) [reply content/ai/concepts/neural-networks/neural-networks.md]: done

**Comment by DevGat718** [comment]
all good

**Comment by DevGat718** [comment]
Hey thanks guys! this was my 1st contribution, when will it be live on docs?

**Pull request merged**
